src/Aggregate/AggregateRepository.php

```diff
@@ -210,14 +221,38 @@ protected function loadFromSnapshotStore(string $aggregateId)
 
         $streamName = $this->determineStreamName($aggregateId);
 
-        $streamEvents = $this->eventStore->loadEventsByMetadataFrom(
-            $streamName,
-            [
-                '_aggregate_type' => $this->aggregateType->toString(),
-                '_aggregate_id' => $aggregateId
-            ],
-            $snapshot->lastVersion() + 1
-        );
+        if ($this->oneStreamPerAggregate) {
+            $streamEvents = $this->eventStore->loadEventsByMetadataFrom(
+                $streamName,
+                $snapshot->lastVersion() + 1
+            );
+        } else {
+            $metadataMatcher = new MetadataMatcher();
+            $metadataMatcher = $metadataMatcher->withMetadataMatch(
+                '_aggregate_type',
+                Operator::EQUALS(),
+                $this->aggregateType->toString()
+            );
+            $metadataMatcher = $metadataMatcher->withMetadataMatch(
+                '_aggregate_id',
+                Operator::EQUALS(),
+                $aggregateId
+            );
+            $metadataMatcher = $metadataMatcher->withMetadataMatch(
+                '_version',
```

Comment:
I know it requires more work, but it would be nice if we rename `_version` to `_aggregate_version` to make the concept and difference between domain event version and event no in the streams explicit. What do you think @prolic ?